scale/job/models.py

```diff
@@ -1481,6 +1481,55 @@ class Meta(object):
         index_together = ['job', 'exe_num']
 
 
+class JobInputFileManager(models.Manager):
+    """Provides additional methods for handleing JobInputFiles"""
+
+    def get_job_input_files(self, job_id, started=None, ended=None, time_field=None, file_name=None, job_input=None):
+        """Returns a query for Input Files filtered on the given fields.
+
+        :param job_id: The job ID
+        :type job_id: int
+        :param started: Query Scale files updated after this amount of time.
+        :type started: :class:`datetime.datetime`
+        :param ended: Query Scale files updated before this amount of time.
+        :type ended: :class:`datetime.datetime`
+        :keyword time_field: The time field to use for filtering.
+        :type time_field: string
+        :param file_name: Query Scale files with the given file name.
+        :type file_name: str
+        :param job_input: The name of the job input that the file was passed into
+        :type job_input: str
+        :returns: The Scale file query
+        :rtype: :class:`django.db.models.QuerySet`
+        """
+
+        job_input_files = JobInputFile.objects.filter(job__id=job_id)
+
+        # Get input_file data from JobInputData model
+        if job_input_files:
+            if job_input:
+                job_input_files = job_input_files.filter(job_input=job_input)
+
+            job_input_file_ids = job_input_files.values_list('input_file__id', flat=True)
+
+        # Reach back to the job_data to get input_file data for legacy jobs
+        else:
+            job_data = Job.objects.get(pk=job_id).get_job_data()
+            job_input_files = job_data.get_input_file_info()
+
+            if job_input:
+                job_input_file_ids = [f_id for f_id, name in job_input_files if name == job_input]
+            else:
+                job_input_file_ids = [f_id for f_id, name in job_input_files]
+
+        files = ScaleFile.objects.filter_files(started=started, ended=ended, time_field=time_field,
+                                               file_name=file_name)
+        files = files.filter(id__in=job_input_file_ids)
+        files = files.order_by('last_modified')
+
+        return files
+
+
```

Comment:
We would like to do this in one (or two for legacy) queries. So apply the filtering of job_id and job_input to the queryset return by ScaleFile.objects.filter_files() and see if any files are returned. If not, then basically do the logic in your else block.